x-pack/plugins/fleet/server/services/files/client.test.ts

```diff
@@ -0,0 +1,360 @@
+
+      await expect(getFleetFilesInstance('from-host').get('123')).resolves.toHaveProperty(
+        'status',
+        'DELETED'
+      );
+    });
+  });
+
+  describe('#create() method', () => {
+    let fileReadable: HapiReadableStream;
+
+    beforeEach(() => {
+      fileReadable = createHapiReadableStreamMock();
+    });
+
+    it('should create a new file with expected metadata', async () => {
+      const createdFile = await getFleetFilesInstance('to-host').create(fileReadable, ['123']);
+
+      expect(esFileClientMock.create).toHaveBeenCalledWith({
+        id: expect.any(String),
+        metadata: {
+          meta: {
+            action_id: '',
+            target_agents: ['123'],
+          },
+          mime: 'application/text',
+          name: 'foo.txt',
+        },
+      });
+
+      expect(createdFile).toEqual({
+        actionId: '123',
+        agents: ['abc'],
+        created: '2022-10-10T14:57:30.682Z',
+        id: '123',
+        mimeType: 'text/plain',
+        name: 'test.txt',
+        sha256: 'e5441eb2bb8a774783d4ff4690153832688bd546c878e953acc3da089ac05d06',
+        size: 1234,
+        status: 'READY',
+      });
+    });
+
+    it('should error is `type` is not `to-host`', async () => {
+      await expect(
+        getFleetFilesInstance('from-host').create(fileReadable, ['123'])
+      ).rejects.toThrow('Method is only supported when `type` is `to-host`');
+    });
+
+    it('should error if `agentIds` is empty', async () => {
+      await expect(getFleetFilesInstance('to-host').create(fileReadable, [])).rejects.toThrow(
+        'FleetFilesClientError: Missing agentIds!'
+      );
+    });
+
+    it('should upload a file and use transform to create hash', async () => {
+      const hashTransform = jest
+        .requireActual('@kbn/files-plugin/server')
+        .createFileHashTransform();
+
+      createFileHashTransformMock.mockReturnValue(hashTransform);
+      await getFleetFilesInstance('to-host').create(fileReadable, ['123']);
+
+      expect(esFile.uploadContent).toHaveBeenCalledWith(fileReadable, undefined, {
+        transforms: [hashTransform],
+      });
+    });
+
+    it('should error if hash was not created', async () => {
+      esFile.data.hash = undefined;
+
+      await expect(getFleetFilesInstance('to-host').create(fileReadable, ['123'])).rejects.toThrow(
+        'FleetFilesClientError: File hash was not generated!'
+      );
+    });
+  });
+
+  describe('#update() method', () => {
+    it('should update file with updates provided', async () => {
+      const file = await getFleetFilesInstance('to-host').update('123', {
+        agents: ['bbb', 'ccc'],
+        actionId: 'aaaaaa',
+      });
+
+      expect(esFile.update).toHaveBeenCalledWith({
+        meta: {
+          action_id: 'aaaaaa',
+          target_agents: ['bbb', 'ccc'],
+        },
+      });
+
+      expect(file).toEqual({
+        actionId: '123',
+        agents: ['abc'],
```

Comment:
The goal here is more to validate that the `update()` method return a `FleetFile` record not necessarily  to check that the data in the return record is correct (only testing in a live env. will be able to actually validate that because we're working with mocks here.

but I can probably enhance the mock to `merge()` in the updates and return the updated record as well. 

Thanks 👍